### PR TITLE
Add user documentation for KEP-2937 to website

### PR DESCRIPTION
**Pull request by dgrove-oss**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Adds user-facing documentation on configurable resource transformations (KEP-2937) to website

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

I'm putting NONE for the release note, because #3026 which is the implementation PR contains the release note text.

Written under the assumption that we'll be merge 3026 in time to make 0.9. 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by kannon92** [comment]
Waiting on https://github.com/kubernetes-sigs/kueue/pull/3026.

/hold

Doc looks good to me though.

**Comment by mimowo** [comment]
/hold cancel 
the impl is merged

**Comment by dgrove-oss** [comment]
I expanded the example of [troubleshooting a pending Job](https://deploy-preview-3175--kubernetes-sigs-kueue.netlify.app/docs/tasks/troubleshooting/troubleshooting_jobs/#pending-workload) to show `resourceRequests`. 

I thought about showing two yaml blocks; one with the feature enabled and one without (due, what was there before), but it seemed like overkill.   I could easily change it if that is preferred though.

File: site/content/en/docs/tasks/troubleshooting/troubleshooting_jobs.md [hunk 1]
```diff
@@ -167,8 +167,24 @@ status:
...
+    resources:
+      cpu:    3
+      Memory: 600Mi
 ```
 
+{{< feature-state state="alpha" for_version="v0.9" >}}
+{{% alert title="Note" color="primary" %}}
+
+`resourceRequests` is a Alpha feature that is not enabled by default.  If the feature is not enabled,
+the status will not contain the detailed description of the resources being requested by the Workload.
```
Comment (mimowo) [inline site/content/en/docs/tasks/troubleshooting/troubleshooting_jobs.md:181]:
```suggestion
`resourceRequests` is a Alpha feature that is not enabled by default.  When the feature is enabled,
the status will contain the detailed description of the resources being requested by the Workload.
```
nit :)

**Review by mimowo — COMMENTED** [review]
LGTM
/approve
non blocking nit, anything to add @tenzen-y ?

File: site/content/en/docs/tasks/manage/administer_cluster_quotas.md [hunk 2]
```diff
@@ -400,3 +400,70 @@ resources:
...
+```yaml
+    resources:
+      requests:
+        cpu: 1
+        memory: 100Gi
+      limits:
+        example.com/gpu-type1: 2
+        example.com/gpu-type2: 1
+```
+will for quota purposes have an effective resource request of:
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/manage/administer_cluster_quotas.md:461]:
```suggestion

The Workload obtains an effective resource requests for quota purposes:

```

File: site/content/en/docs/tasks/manage/administer_cluster_quotas.md [hunk 3]
```diff
@@ -400,3 +400,70 @@ resources:
...
+on a transformed set of resources requirements without changing
+the resource requests and limits that the Pods created by the Workload
+will present to the Kubernetes Scheduler when the Workload is admitted by
+a ClusterQueue. Customizations are defined by specifying resource transformations
+in the Kueue configuration as a cluster-level setting.
+
+{{< feature-state state="alpha" for_version="v0.9" >}}
+{{% alert title="Note" color="primary" %}}
+
+`transformations` is a Alpha feature that is not enabled by default.
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/manage/administer_cluster_quotas.md:418]:
```suggestion
`transformations` is an Alpha feature that is not enabled by default.
```

File: site/content/en/docs/tasks/troubleshooting/troubleshooting_jobs.md [hunk 4]
```diff
@@ -167,8 +167,24 @@ status:
...
+    name: main
+    resources:
+      cpu:    3
+      Memory: 600Mi
 ```
 
+{{< feature-state state="alpha" for_version="v0.9" >}}
+{{% alert title="Note" color="primary" %}}
+
+`resourceRequests` is a Alpha feature that is not enabled by default.  When the feature is enabled,
```
Comment (tenzen-y) [inline site/content/en/docs/tasks/troubleshooting/troubleshooting_jobs.md:180]:
```suggestion
`resourceRequests` is an Alpha feature that is not enabled by default.  When the feature is enabled,
```

**Review by tenzen-y — COMMENTED** [review]
Mostly lgtm.
Thanks.

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/approve
/hold cancel

**Pull request merged**